### PR TITLE
capture(cdc): always close processor if etcd sesison is broken

### PR DESCRIPTION
**Pull request by hicqu**
### What problem does this PR solve?

Issue Number: ref #8657 

### What is changed and how it works?

Without the PR if a capture's etcd session is broken, `processor.Close` won't be called correctly. It will leak some resources of `processor`. For now it won't cause critical results but we should fix the behavior, which will make code more clear.

#### Tests 

 - Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/capture/capture.go [hunk 1]
```diff
@@ -599,7 +599,7 @@ func (c *captureImpl) AsyncClose() {
 	c.captureMu.Lock()
 	defer c.captureMu.Unlock()
 	if c.processorManager != nil {
-		c.processorManager.AsyncClose()
+		c.processorManager.Close()
```
Comment (asddongmen) [inline cdc/capture/capture.go:602]: May be we also need to change the `capture.AsyncClose` to  `capture.Close`.

File: cdc/processor/manager.go [hunk 2]
```diff
@@ -183,16 +183,13 @@ func (m *managerImpl) closeProcessor(changefeedID model.ChangeFeedID, ctx cdcCon
...
-// AsyncClose sends a signal to Manager to close all processors.
-func (m *managerImpl) AsyncClose() {
-	timeout := 3 * time.Second
-	ctx, cancel := context.WithTimeout(context.TODO(), timeout)
-	defer cancel()
-	done := make(chan error, 1)
-	err := m.sendCommand(ctx, commandTpClose, nil, done)
-	if err != nil {
-		log.Warn("async close failed", zap.Error(err))
+// Close the manager itself and all processors. Can't be called with `Tick`.
```
Comment (asddongmen) [inline cdc/processor/manager.go:186]:
```suggestion
// Close the manager itself and all processors. 
// Note: This method must not be called with `Tick`. Please be careful.
```

**Review by asddongmen — APPROVED** [review]

File: cdc/capture/capture.go (hunk 1 above)
Comment (hicqu) [reply cdc/capture/capture.go:602]: We will consider it, when examine `owner.AsyncStop`.

File: cdc/processor/manager.go [hunk 3]
```diff
@@ -159,11 +159,11 @@ func (m *managerImpl) Tick(stdCtx context.Context, state orchestrator.ReactorSta
 	return state, nil
 }
 
-func (m *managerImpl) closeProcessor(changefeedID model.ChangeFeedID, ctx cdcContext.Context) {
+func (m *managerImpl) closeProcessor(changefeedID model.ChangeFeedID) {
```
Comment (CharlesCheung96) [inline cdc/processor/manager.go:162]: Would you please add some comment to explain that this function may stuck the tick function?

**Review by CharlesCheung96 — APPROVED** [review]

File: cdc/processor/manager.go (hunk 3 above)
Comment (hicqu) [reply cdc/processor/manager.go:162]: I add comment for `processor.Tick` and `processor.Close`, which can be called in `processor.Manager.Tick`.

**Pull request merged**